app/src/handlers/aggregate/global/GlobalFactionCombatAggregate.ts

```diff
@@ -0,0 +1,123 @@
+
+        // Create initial record if doesn't exist
+        if (!await this.factory.model.exists({
+            world: event.alert.world,
+        })) {
+            await this.insertInitial(event);
+        }
+
+        const documents = [];
+
+        // Increment attacker faction kills
+        if (!event.isTeamkill && !event.isSuicide) {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/restrict-template-expressions
+            const attackerKillKey = `${FactionUtils.parseFactionIdToShortName(event.attackerFaction)}.kills`;
+            documents.push(
+                {$inc: {[attackerKillKey]: 1}},
+                {$inc: {['totals.kills']: 1}},
+            );
+        }
+
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/restrict-template-expressions
+        const victimDeathKey = `${FactionUtils.parseFactionIdToShortName(event.characterFaction)}.deaths`;
+        documents.push(
+            {$inc: {[victimDeathKey]: 1}},
+            {$inc: {['totals.deaths']: 1}},
+        );
+
+        if (event.isTeamkill) {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/restrict-template-expressions
+            const teamKillKey = `${FactionUtils.parseFactionIdToShortName(event.attackerFaction)}.teamKills`;
+            documents.push(
+                {$inc: {[teamKillKey]: 1}},
+                {$inc: {['totals.teamKills']: 1}},
+            );
+        }
+
+        if (event.isSuicide) {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/restrict-template-expressions
+            const suicideKey = `${FactionUtils.parseFactionIdToShortName(event.characterFaction)}.suicides`;
+            documents.push(
+                {$inc: {[suicideKey]: 1}},
+                {$inc: {['totals.suicides']: 1}},
+            );
+        }
+
+        if (event.isHeadshot) {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/restrict-template-expressions
+            const attackerHeadshotKey = `${FactionUtils.parseFactionIdToShortName(event.attackerFaction)}.headshots`;
+            documents.push(
+                {$inc: {[attackerHeadshotKey]: 1}},
+                {$inc: {['totals.headshots']: 1}},
+            );
+        }
+
+        // It's an old promise sir, but it checks out (tried Async, doesn't work with forEach)
+        documents.forEach((doc) => {
+            void this.factory.model.updateOne(
+                {world: event.alert.world},
+                doc,
+            ).catch((err) => {
+                // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+                GlobalFactionCombatAggregate.logger.error(`Updating GlobalFactionCombatAggregate Error! ${err}`);
+            });
+        });
+
+        return true;
+    }
+
+    public async insertInitial(event: DeathEvent): Promise<boolean> {
```

Comment:
Same method with slight variations, maybe a candidate to be added as a mixin to remove some of the code duplication?

Reply:
Quite possibly. Will likely refactor at a later date